doc/source/debug.rst

```diff
@@ -0,0 +1,191 @@
+    hub-deployment-3311438805-xnfvp     1/1       Running        0          2m
+    jupyter-choldgraf                   0/1       ErrImagePull   0          25s
+    proxy-deployment-1227971824-mn2wd   1/1       Running        0          5h
+
+Here we can see the two JupyterHub pods, as well as a single user pod. Note
+that all user pods will begin with ``jupyter-``.
+
+In particular, keep an eye on the ``STATUS`` column. If a given
+pod contains something other than ``Running``, then something may be wrong.
+
+In this case, we can see that our user's pod is in the ``ErrImagePull`` state.
+This generally means that there's something wrong with the Docker image that
+is defined in ``singleuser`` in our helm chart. Let's dig further...
+
+``kubectl describe pod``
+~~~~~~~~~~~~~~~~~~~~~~~~
+To see more detail about the state of a specific pod, use the following
+command::
+
+    kubectl --namespace=jhub describe pod <POD_NAME>
+
+This will output several pieces of information, including configuration and
+settings for the pod. The final section you'll see is a list of recent
+events. These can be particularly informative, as often an error will
+show up in this section.
+
+**Real world scenario:** In our case, one of the lines in the events page
+displays an error::
+
+      $ kubectl --namespace=jhub describe pod jupyter-choldgraf
+      ...
+      2m            52s             4       kubelet, gke-jhubtest-default-pool-52c36683-jv6r        spec.containers{notebook}       Warning         Failed           Failed to pull image "jupyter/scipy-notebook:v0.4": rpc error: code = 2 desc = Error response from daemon: {"message":"manifest for jupyter/scipy-notebook:v0.4 not found"}
+      ...
+
+It seems there is indeed something wrong with the Docker image. Let's confirm
+this by getting another view on the events that have transpired in the pod.
+
+``kubectl logs``
+~~~~~~~~~~~~~~~~
+If you only want to see the latest logs for a pod, use the following command::
+
+    kubectl --namespace=jhub logs <POD_NAME>
+
+This will output a list of the recent events for the pod. Parse these logs
+to see if something is generating an error.
+
+**Real world scenario:** In our case, we get this line back::
+
+    $ kubectl --namespace=jhub logs jupyter-choldgraf
+    Error from server (BadRequest): container "notebook" in pod "jupyter-choldgraf" is waiting to start: trying and failing to pull image
+
+Now we are sure that something is wrong with our Docker file. Let's check
+our ``config.yaml`` file for the section where we specify the user's
+Docker image. Here we see our problem::
+
+  singleuser:
+  image:
+      name:
+          jupyter/scipy-notebook
+
+We haven't specified a ``tag`` for our Docker image! Not specifying a tag
+will cause it to default to ``v0.4``, which isn't what we want and is causing
+the pod to fail.
+
+To fix this, let's add a tag to our ``config.yaml`` file::
+
+  singleuser:
+  image:
+      name:
+          jupyter/scipy-notebook

```

Comment:
Canonically, it it should be 'name: <something>' and 'tag: <something>' rather than across multiple lines